src/mail.sh

```diff
@@ -0,0 +1,125 @@
+# This file handles all the interactions of git send-email. It handles its
+# configuration proccess to facilitate the sending of patches via e-mail.
+
+include "$KW_LIB_DIR/kwlib.sh"
+include "$KW_LIB_DIR/kwio.sh"
+include "$KW_LIB_DIR/kw_config_loader.sh"
+
+# Hash containing user options
+declare -gA options_values
+
+function kw_mail()
+{
+  parse_mail_options "$@"
+  if [[ "$?" -gt 0 ]]; then
+    complain "${options_values['ERROR']}"
+    return 22 # EINVAL
+  fi
+
+  if [[ "${options_values['SETUP']}" == 1 ]]; then
+    mail_setup
+    exit
+  fi
+
+  return 0
+}
+
+# This function deals with configuring the necessary options to use
+# the git send-email tool to send patches by email
+function mail_setup()
+{
+  if ! git rev-parse --is-inside-work-tree &> /dev/null && [[ -z "${options_values['GLOBAL']}" ]]; then
+    complain 'Not in a git repository, aborting setup!'
+    say 'To apply settings globally rerun with "--global" flag.'
+    exit 1 # EPERM
+  fi
+
+  for option in "${!options_values[@]}"; do
+    if [[ "$option" =~ ^SMTP.* ]] && [[ -n "${options_values[$option]}" ]]; then
+      validate_encryption "$option"
+      # TODO change to eval
+      echo "git config ${options_values['GLOBAL']}${options_values[$option]}"
+    fi
+  done
+
+  return 0
+}
+
+# encryption must be 'ssl' or 'tls' this validates this option
```

Comment:
Use the default comment style (see build or deploy) and improve this comment because it does not provide a good explanation.